### DISCUSSION TITLE
Effects of the AI Software - Stops working after some months

### DISCUSSION THREAD

**Discussion by Wollli** [jomjol/AI-on-the-edge-device#3078] [General]
Hello,

I'm using this software now over several years. I run a water counter and a gas counter. ( V 15.4.0)
The software is normal very good.

What I'm observing is this: After a while of usage (some months or so) the gas counter stopped getting new values.

First I inspected the hardware, but this was not changed and working, also not run hot (cellar = 18 deg. C)
- The live image function did work fine. (-> Camera ok, -> Illumination ok)
- Log files could be written  (-> SD Card ok)

Then in config file I set 
  setupmode = false
  autostart = false 
and restartet the system.

I tried to take a new reference image, but this was not possible, LOG File :

> [0d00h01m04s] 2024-05-20T16:26:49 <ERR> [OTA FILE] Failed to stat file: /sdcard/config/reference.jpg!

I tried it more than once, impossible. I deleted the reference.jpg, but also after this same problem.

Then I remembered I had a similar problem about 1 year ago on the water counter:

**Now I looked for the latest release (now 15.7.0) and made an OTA update -> The release Info did not show
of this problem , but it helped. After the update I only had to take a new reference image and the system
was running again. No need to take new reference markers or something else.** 

I don't think that THIS Version did solve the problem, but the update itself. 
Is it possible that during the update some memory is reseted or something else what helps ?
Last time when I observed the effect, is was by using older versions..

What I did not try is to update to the same version...

Best regards
   Wolli

**Comment by SybexX** [comment]
fix(tflite): Fix memory leaks in tflite integration (https://github.com/jomjol/AI-on-the-edge-device/pull/2842)
